### ISSUE TITLE
[FEATURE] scribe.rip alternative to medium.com

### ISSUE THREAD

**Issue by benbusby** [benbusby/whoogle-search#455]
**Describe the feature you'd like to see added**
Allow [scribe.rip](https://scribe.rip) to replace all `medium.com` and `*.medium.com` results

**Additional context**
Should function the same as `twitter.com` -> `nitter.net`, `instagram.com` -> `bibliogram.art/u`, etc, but with the exception that it needs to replace the full domain (including subdomain) for every matching result, not just the `medium.com` portion.

**Comment by benbusby** [comment]
Anyone interested contributing for hacktoberfest (or otherwise just wanting to contribute) -- this should be a pretty easy issue to tackle that adds a lot of value.

The main things to do would be:
- Switch the behavior of `app/utils/results.py -> get_site_alt()` to use `urlparse(...).netloc` to replace the domain path (rather than just doing a simple string replacement) in order to encapsulate subdomains as well.
- Add config settings for setting/replacing the scribe.rip resource to use for replacing Medium links. It'd need to include a `WHOOGLE_ALT_MD` env var, or something similar. 

If interested, please assign yourself this issue to mark that it's in progress. Otherwise I'll likely end up tackling this fairly soon (as I often get frustrated with the Medium paywalls).

**Comment by vacom13** [comment]
I am interested in contributing.  Could you guide me as I am fairly new to all this?

**Comment by benbusby** [comment]
@vacom13 New to the project itself? Or new to contributing on github?

Either way, this feature would involve the changes:

- Updating the `get_site_alt()` method in `app/utils/results.py` with something like the following:
```python
for site_key in SITE_ALTS.keys():
  if site_key not in link:
      continue
  
  # Need to replace full hostname with alternative to encapsulate
  # subdomains as well
  hostname = urlparse.urlparse(link).hostname
  
  link = link.replace(hostname, SITE_ALTS[site_key])
  for prefix in SKIP_PREFIX:
      link = link.replace(prefix, '//')
  break

return link
```

Haven't tested that, but should work. Then updating the `SITE_ALTS` mapping in the same class to include a `medium.com` that maps to a `WHOOGLE_ALT_MD` environment variable or something like that with a default value of `scribe.rip`. 

You'd also need to update the documentation in the README with the new environment variable name and purpose, and include it in `whoogle.env`, the Dockerfile, `app.json`, and `docker-compose.yml`. It sounds like a lot but it's all pretty quick copy/paste stuff based on existing environment variables that you can reference like `WHOOGLE_ALT_TW` and `WHOOGLE_ALT_IG`. 

Let me know if this makes sense. Happy to help further if needed!

**Comment by vacom13** [comment]
Well I am new to github. Just came across this rep. Felt maybe should try to contribute to it. Thank you. I will work on it!

**Comment by benbusby** [comment]
Awesome. I'll review the PR this weekend and let you know if there's anything else needed. Thanks for taking care of this!

**Comment by benbusby** [comment]
Merged via #463 -- thanks again @vacom13, great work. 

I'm slowly trying to mark the more straightforward issues with the label "good first issue". Keep an eye out for those and let me know if you want to contribute to any that you see.

**Comment by vacom13** [comment]
@benbusby Sure thing!
